pkg/gitprovider/bitbucketserver_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitprovider
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type BitbucketServerGitProviderTestSuite struct {
+	gitProvider *BitbucketServerGitProvider
+	suite.Suite
+}
+
+func NewBitbucketServerGitProviderTestSuite() *BitbucketServerGitProviderTestSuite {
+	baseApiUrl := "https://bitbucket.example.com"
+	return &BitbucketServerGitProviderTestSuite{
+		gitProvider: NewBitbucketServerGitProvider("username", "token", &baseApiUrl),
+	}
+}
+
+func (b *BitbucketServerGitProviderTestSuite) TestParseStaticGitContext_PR() {
+	prUrl := "https://bitbucket.example.com/rest/api/latest/projects/PROJECT_KEY/repos/REPO_NAME/pull-requests/1"
```

Comment:
@abhishek818 seems that you tested the wrong URLs here.

![Screenshot 2024-06-17 at 10 07 41](https://github.com/daytonaio/daytona/assets/26512078/20722bfe-740f-4040-a51d-da8c66cb25b4)

As you can see here, when I attempt to clone from a PR URL, the creation process fails. Take note that the PR URL does not have `rest/api/latest` as the prefix. This is true for other URLs in the browser as well. Please update your tests and git parsing to correctly reflect this.

Reply:
@Tpuljak
I have kept the `'rest/api/<version>'` as optional string while parsing.
`re := regexp.MustCompile(`(https?://[^/]+)(?:/rest/api/[^/]+)?.. `
I just rechecked the parsing test cases by removing this string to ensure if they pass.

Seems issue you are facing is something else - "Project test does not exist" ? 
else for parsing issues, either cli should have abruptly exit or thrown "Could not extract project key and repo name from URL: %s"

Reply:
Can you confirm that you are able to create a repo using a PR URL?

Reply:
@Tpuljak Nah, not able to repro due to auth issues..
 
```
/workspace/kane (feat_bitbucket_self_hosting) $ git clone ssh://git@7990-abhishek818-daytona-n2970r706kk.ws-us114.gitpod.io/dayt/daytona.git
Cloning into 'daytona'...
Connection closed by 34.168.189.139 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

gitpod /workspace/daytona (feat_bitbucket_self_hosting) $ ./daytona create https://7990-abhishek818-daytona-n2970r706kk.ws-us114.gitpod.io/projects/DAYT/repos/daytona/browse
FATA[0000] failed to get repository: Status: 401 Unauthorized, Body:  ParseError: unexpected end of JSON input 
```

Reply:
@abhishek818 from what I can see you don't have ssh access setup correctly in your workspace so `git clone` doesn't work.

Regarding the second line, did you add your token with the appropriate scopes and did you add the correct API URL?

Reply:
atleast .git and browse urls are going thru (in previous reply), else any incorrect hostnames/urls throw |

```
git clone ssh://git@random_hostname/daytona.
git
Cloning into 'daytona'...
ssh: Could not resolve hostname random_hostname: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Reply:
> @abhishek818 from what I can see you don't have ssh access setup correctly in your workspace so `git clone` doesn't work.
> 
> Regarding the second line, did you add your token with the appropriate scopes and did you add the correct API URL?

Yes, everything's correct, with proper scopes, have tried with both http and ssh ways , generating public key in terminal and adding as ssh key in gyi, but nothing works

Reply:
@abhishek818 unfortunately, I can't provide more help with this as this seems like an issue with your workspace or Bitbucket Server setup. It's hard to debug this issues like this and this PR thread is not really a place for it.

I suggest you either try to setup the Server another way, generate a new token with required scopes or try to find an alternative way to test this PR.

I'm putting it back to draft. I won't close it since I see you're close to the finish line but please reopen it when you had a chance to properly test all the functionality.

Reply:
hey you were able to execute with the PR url [earlier](https://github.com/daytonaio/daytona/pull/606#discussion_r1635947522), the only issue was a null pointer in the api response. Since then, no code changes relative to PR url was done.

Reply:
@abhishek818 I am able to create a workspace a PR using the TUI (`daytona create`) but the issue I encountered now is regarding to `daytona create PR_URL`.

Reply:
fixed.